frontend/packages/console-dynamic-plugin-sdk/src/api/api-types.ts

```diff
@@ -33,7 +33,7 @@ export type WatchK8sResources<R extends ResourcesObject> = {
 };
 
 export type UseK8sWatchResource = <R extends K8sResourceCommon | K8sResourceCommon[]>(
-  initResource: WatchK8sResource,
+  initResource: WatchK8sResource | null,
```

Comment:
Just curious, why is this needed?

Reply:
In BuildConfigFormPage I used this code to fetch and watch the BuildConfig only if a name is defined.

```tsx
  const isNew = !name || name === '~new';
  const [watchedBuildConfig, loaded, loadError] = useK8sWatchResource<BuildConfig>(
    isNew
      ? null
      : {
          kind: BuildConfigModel.kind,
          name,
          namespace,
        },
  );
```

`useK8sWatchResource` already supported to skip fetching if the parameter is null but the type definition doesn't allow this.